### ISSUE TITLE
[ideas] Dashboard, exit confirmation, better resuming

### ISSUE THREAD

**Issue by fuzzah** [AFLplusplus/AFLplusplus#274]
Just lost my status screens with few billions of execs :(
So, my ideas:
1. Can we store all the fuzzer stats as shown in status screen, and then restore it when resuming (with both AFL_AUTORESUME and '-i' option) ? If it somehow affects or depends on inner state of AFL++ instance, can we restore this inner state too?
2. Can we prevent random Ctrl+C presses for example, by asking to confirm? You know, there may be lots of opened windows, tty multiplexer sessions and so on, anything can happen.
3. What if AFL++ had some sort of dashboard / control center available via web interface? It could mimic afl-whatsup and afl-plot in some ways. It could be used for much better monitoring of current fuzzing job (and maybe also show some analysis of previous runs too). It could also be used for showing separate status screens, restarting instances with provided command lines, changing input test cases, etc .. Most importantly, it would provide a way to access all the stats from any device capable of web surfing, so that you could monitor & control your fuzzing jobs while walking your dog :D

**Comment by vanhauser-thc** [comment]
well the fuzzer_stats could be loaded on resume and variables set accordingly. This is however nothing important. you lost something that looked cool, but you lost nothing of real value :)
If you would like to have that in, send a PR, I think everybody else has something with higher priority on their plate

**Comment by tushargoyl123** [comment]
Hi this is Tushar , I want to work on this issue , I am new to open source Can u guide me through??

**Comment by vanhauser-thc** [comment]
@tushargoyl123 do have you a strong experience in coding C/C++?

**Comment by tushargoyl123** [comment]
> @tushargoyl123 do have you a strong experience in coding C/C++?

Yes I know it very well...

**Comment by vanhauser-thc** [comment]
@tushargoyl123 the only relevant feature request for this repo is the first one IMHO (2 is undesired, 3 is out of scope and rather an own project):

```
Can we store all the fuzzer stats as shown in status screen, and then restore it when resuming (with both AFL_AUTORESUME and '-i' option) ? If it somehow affects or depends on inner state of AFL++ instance, can we restore this inner state too?
```

basically what has to be done is that if `afl-fuzz` is started in resume mode, then the existing "$out_dir/fuzzer_stats" file has to be loaded, the values read in and assigned to the internal variables. pretty easy and fast to do - if you know what you are doing.

but the hard part is to understand how afl++ works. I recommend reading the README.md fully, then try out some of the tutorials that are mentioned there. after that take a look at `src/afl-fuzz-stats.c` - which should then be familiar what is happening there. If you have questions then, just write them here.

**Comment by tushargoyl123** [comment]
@vanhauser-thc  ok once I go through all the things I will let you know...

**Comment by tushargoyl123** [comment]
#272

**Comment by NishantPuri99** [comment]
Hi! Can I try this issue? I'm good with C/C++ and have a very basic knowledge of Fuzzing.

**Comment by vanhauser-thc** [comment]
@NishantPuri99 I am not sure if @tushargoyl123 is working on this already.

Another open issue is that afl++ compiles and works on Android - if that is something one of you are interested in doing.

**Comment by NishantPuri99** [comment]
Hi again, I'll work on that then, could you please guide me regarding it?

**Comment by tushargoyl123** [comment]
@vanhauser-thc hi i am so sorry my practical exams were going on in my college I have set up the chapel environment, have run some basic commands now coming on the issue I need to learn a little bit from you like i can't understand the meaning of status screen(here) Also tell me how would you approach to solve this issue Would be a great help... Will be back with this issue solved very soon :)

**Comment by vanhauser-thc** [comment]
@tushargoyl123 all your questions on status screen, basic command, etc. are answered in the documentation, e.g, the main README.md and docs/status_screen.md

**Comment by mykter** [comment]
Looks like this can probably be closed now that #728 is merged?

**Comment by domenukk** [comment]
Right, this and statsd (#684 & #702) should do the trick
